clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -408,7 +408,7 @@ public KafkaProducer(Properties properties, Serializer<K> keySerializer, Seriali
                 this.metadata = metadata;
             } else {
                 this.metadata =
-                    new Metadata(retryBackoffMs, config.getLong(ProducerConfig.METADATA_MAX_AGE_CONFIG), true, true,
+                    new Metadata(retryBackoffMs, config.getLong(ProducerConfig.METADATA_MAX_AGE_CONFIG), config.getBoolean(ProducerConfig.ALLOW_AUTO_CREATE_TOPICS_CONFIG), true,
```

Comment:
Could you make a follow up fix in the server side to not to "log" (revert your previous commit) when topic cannot be created? Now we turned off auto topic creation, producer/consumer could keep retrying and flood the log.

Reply:
Yes. I will